Improve usability of gRPC probes in Kubernetes

Sgitario opened the pull request:
Plus, I'm adding a new guide about the usage of gRPC services in Kubernetes. 
Relates to https://github.com/quarkusio/quarkus/issues/33219#issuecomment-1540067363

cescoffier commented on docs/src/main/asciidoc/grpc-kubernetes.adoc:
```diff
@@ -0,0 +1,81 @@
+----
+
+By default, the Quarkus Kubernetes will expose the HTTP port with name `http` which is bound to the Quarkus HTTP server, not the gRPC service. To expose the gRPC server instead, set the `quarkus.kubernetes.ingress.target-port=grpc` property in your application.properties:
+
```
How does that work with the new server where both http and gRPC are served on the same port?

Sgitario replied:
Can you share more details about this? But I guess that if both HTTP and gRPC are handing by the same port, it will work fine using the default `http` port name. 
Though, if the port name `grpc` won't be generated any longer, we might need to revisit this statement.

cescoffier replied:
With the old gRPC server, http and gRPC are served on different ports.
With the new gRPC server, HTTP and gRPC are served on the same port (which is very likely going to be https). 

Both mode are supported at the moment.

cescoffier replied:
@alesj do you think we could add tests for both cases?

Sgitario replied:
I took a deeper look into this that was implemented as part of https://github.com/quarkusio/quarkus/pull/28654.
So, by default, it creates two servers: one for HTTP and another one for gRPC. 
If `quarkus.grpc.server.use-separate-server` is `false`, then the gRPC port won't be listening. 

Note that the `grpc` port is still bound to the generated Deployment/Service resources even though `quarkus.grpc.server.use-separate-server` is false. Because `quarkus.grpc.server.use-separate-server` is a runtime property, this might make sense. So, ultimately the user would need to select the target port (either `http` or `grpc`) of the generated resource. 

We could again follow a best-effort approach, so if users use `quarkus.grpc.server.use-separate-server=false` at build time, we could remove the binding of the `grpc` port in the generated resources and use `http` by default as the target port. Wdyt @cescoffier ?

cescoffier replied:
+1 or having an explicit property on the kubernetes side.

cescoffier commented on docs/src/main/asciidoc/grpc-kubernetes.adoc:
```diff
@@ -0,0 +1,81 @@
+
+[source,shell]
+----
+$ mvn clean package
```
May need the tabs with gradle and cli

Sgitario replied:
do you want me to update the rest of gRPC guides with this too ?

cescoffier replied:
If you can, that would be great.

I have a task somewhere about restructuring the whole gRPC doc. But didn't have the time to complete it.

Sgitario replied:
PR updated with this change.

iocanel commented on extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ProbeConfig.java:
```diff
@@ -54,6 +54,12 @@ public class ProbeConfig {
+     * If enabled and `grpc-action` is not provided, it will use the generated service name and the gRPC port.
+     */
+    @ConfigItem(defaultValue = "false")
+    boolean grpcActionEnabled;
```
I am wondering  if there are ways to infer this configuration value using a  build item or something like that.
I would like to avoid having `grpcAction` and `grpcActionEnabled` as it can be confusing.

If this is not possible then we need to use a name that clearly suggests that by setting this to `true` the grpcAction will be infered. But I would prefer if we could somehow completly avoid having a new property.

iocanel replied:
For now we can leave it as is. I created: https://github.com/quarkusio/quarkus/issues/33583 as a general solution / approach for such cases.

iocanel reviewed (changes requested):
Overall, looks good I'd like us to change `grpcActionEnabled` thing. See comments for details.

Sgitario committed:
Improve usability of gRPC probes in Kubernetes

Plus, I'm adding a new guide about the usage of gRPC services in Kubernetes.
Relates to https://github.com/quarkusio/quarkus/issues/33219#issuecomment-1540067363

Sgitario commented:
@cescoffier PR updated with your two change requests. 
@iocanel my question still stands https://github.com/quarkusio/quarkus/pull/33437#discussion_r1200266488

iocanel reviewed (approved):
lgtm
